examples/CMakeLists.txt

```diff
@@ -31,11 +34,12 @@ target_include_directories(publisher_component PUBLIC
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
   $<INSTALL_INTERFACE:include>
 )
-ament_target_dependencies(publisher_component
-  rclcpp
-  rclcpp_components
-  std_msgs
+target_link_libraries(publisher_component PUBLIC
+  ${std_msgs_TARGETS}
+  rclcpp::rclcpp
```

Comment:
So, if now we use proper CMake targets as public dependencies, we can probably get rid of `ament_target_dependencies` also in the test build confi?